observability/src/test/java/io/grpc/observability/interceptors/LogHelperTest.java

```diff
@@ -448,16 +566,46 @@ public void logTrailer() throws Exception {
           methodName,
           statusDescription.getCode().toStatus(),
           nonEmptyMetadata,
+          HEADER_LIMIT,
           EventLogger.LOGGER_CLIENT,
           rpcId,
           peerAddress);
       verify(sink).write(
           base.toBuilder()
               .clearStatusMessage()
-              .build());
+              .build(),
+          locationTags, customTags);
     }
   }
 
+  @Test
+  public void alwaysLoggedMetadata_grpcTraceBin() throws Exception {
```

Comment:
Already commented on the relevance of always-logged and never-logged metadata. What's the requirement?